### PR TITLE
Store fileTypeColors as Vuex global variable

### PR DESCRIPTION
**Pull request by vig42**
Partially addresses #1268 

```
Currently we are passing fileTypeColors as a prop across the components.

As we have now implemented Vuex, let's store fileTypeColors as a Vuex
global variable to simplify the structure.
```

### CODE REVIEW

**Review by jamessspanggg — APPROVED** [review]

**Review by fzdy1914 — APPROVED** [review]
LGTM

File: frontend/src/static/js/v_summary_charts.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ window.vSummaryCharts = {
     },
 
     ...Vuex.mapState(['mergedGroups']),
+    ...Vuex.mapState(['fileTypeColors']),
```
Comment (fzdy1914) [inline frontend/src/static/js/v_summary_charts.js:34]: Can this statement be merged with the previous one?

Comment (vig42) [reply frontend/src/static/js/v_summary_charts.js]: Yes, I have merged it.

File: frontend/src/static/js/v_authorship.js [hunk 2]
```diff
@@ -121,7 +121,7 @@ window.vAuthorship = {
       }
 
       if (!this.info.fileTypeColors) {
-        this.$root.$emit('restoreFileTypeColors', this.info);
+        this.info.fileTypeColors = this.$store.state.fileTypeColors;
```
Comment (fzdy1914) [inline frontend/src/static/js/v_authorship.js:124]: image

I believe there is no need to store fileTypeColors in the info anymore. Let's set it to a computed map state value just like how we do it in `vSummaryCharts`

**Review by 0blivious — APPROVED** [review]

**Review by fzdy1914 — APPROVED** [review]

**Review by jamessspanggg — COMMENTED** [review]
Seems like the file types are not restored after reloading the page, is it related to ur PR?

filetype

Don't think so, I'm getting the same issue on master too. Specifically, if you select 2 or more file types, it does not remember your selection when you refresh. Strangely, if you only select 1 type (e.g. only selecting frontend), it remembers your selection. Issue is present on both Chrome and Firefox.

**Comment by vig42** [comment]
> 
> 
> Seems like the file types are not restored after reloading the page, is it related to ur PR?
> 
> filetype


Don't think so, I'm getting the same issue on master too. Specifically, if you select 2 or more file types, it does not remember your selection when you refresh. Strangely, if you only select 1 type (e.g. only selecting frontend), it remembers your selection. Issue is present on both Chrome and Firefox.

**Comment by Tejas2805** [comment]
From what I remember, we don't restore `fileTypes` currently. Based on the tests I wrote, I think this is the case.

**Comment by jamessspanggg** [comment]
> Don't think so, I'm getting the same issue on master too. Specifically, if you select 2 or more file types, it does not remember your selection when you refresh. Strangely, if you only select 1 type (e.g. only selecting frontend), it remembers your selection. Issue is present on both Chrome and Firefox.

Thanks for investigating, perhaps can file this as an issue?

**Review by jamessspanggg — APPROVED** [review]

**Pull request merged**